### PR TITLE
EARTH-216: Accessibility Improvements From AMP Report

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW/

# Summary
- Adds title and role to SVG Elements
- Adds role=navigation to collapsible menu 
- Adds LI elements to the UL element in the collapsible menu (was just rendering the a tag before)
- Added many more aria-labels to correctly announce the right thing when tab navigating

# Needed By (Date)
- Before the end of the sprint

# Urgency
- Low

# Steps to Test

1. Check out this branch
2. Clear caches
3. Review /pattern page source for markup changes
4. Enable voice over and tab through the page evaluating the spoken feedback.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
@pookmish,

Ok, I really am done with this now.

File: patterns/molecules/photo-social/photo-social.html.twig [hunk 1]
```diff
@@ -4,10 +4,16 @@
  * Photo Social pattern.
  */
 #}
+{% if title is not empty %}
+  {% set arialabel = title|render|striptags|preg_replace(":", "") %}
```
Comment (pookmish) [inline patterns/molecules/photo-social/photo-social.html.twig]: interesting. `:` has some issues?

File: src/TwigExtension/StanfordComponentsTwigExtension.php [hunk 2]
```diff
@@ -160,6 +162,31 @@ public function firstChar($item) {
...
+  public function getImgAlt($item) {
+    $item = render($item);
+
+    if (strpos($item, '<img') !== FALSE) {
+      $dom = new \DOMDocument('1.0', 'UTF-8');
+      libxml_use_internal_errors(TRUE);
+      $dom->loadHTML($item);
+      /** @var \DOMNodeList $images */
+      $images = $dom->getElementsByTagName('img');
+      return $images->item(0)->getAttribute('alt');
```
Comment (pookmish) [inline src/TwigExtension/StanfordComponentsTwigExtension.php]: Should this check if images is empty?

File: patterns/molecules/photo-social/photo-social.html.twig (hunk 1 above)
Comment (sherakama) [reply patterns/molecules/photo-social/photo-social.html.twig]: Yeah, something drupal/twig related in which it only outputs the parts after the colon.

File: src/TwigExtension/StanfordComponentsTwigExtension.php (hunk 2 above)
Comment (sherakama) [reply src/TwigExtension/StanfordComponentsTwigExtension.php]: Hmm. probably. I'll revise.

**Comment by sherakama** [comment]
@pookmish revised.

File: src/TwigExtension/StanfordComponentsTwigExtension.php [hunk 3]
```diff
@@ -160,6 +162,38 @@ public function firstChar($item) {
...
+      /** @var \DOMNodeList $images */
+      $images = $dom->getElementsByTagName('img');
+
+      // Can only act on something that exists.
+      if (empty($images)) {
+        return;
+      }
+
+      $text = $images->item(0)->getAttribute('alt');
+
```
Comment (pookmish) [inline src/TwigExtension/StanfordComponentsTwigExtension.php]: missing return

Comment (pookmish) [reply src/TwigExtension/StanfordComponentsTwigExtension.php]: I didnt notice your original had `strpos($item, '<img'))` which kinda did the same as i suggested originally.

**Pull request merged**